src/engine/strat_engine/util.rs

```diff
@@ -54,3 +55,21 @@ pub fn set_uuid(devnode: &Path, uuid: Uuid) -> EngineResult<()> {
                     .arg(&devnode),
                 &format!("Failed to set UUID for filesystem {:?}", devnode))
 }
+
+/// Lookup the WWN from the udev db using the device node eg. /dev/sda
+pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
+    let context = libudev::Context::new()?;
+    let mut enumerator = libudev::Enumerator::new(&context)?;
+    enumerator.match_subsystem("block")?;
+    enumerator.match_property("DEVTYPE", "disk")?;
+
+    let result = enumerator
```

Comment:
Probably unnecessary to assign to ```result``` and then immediately return it.

Reply:
Code will not compile it, get 2 borrow errors.

Reply:
Has bug been filed w/ clippy yet? If not, probably worth it.

Reply:
_ Perhaps_ instead of ```String::from(x)```, ```x.to_owned()``` would avoid error. It might ease the borrow checker's difficulties. But I wouldn't bet on it.

Reply:
Tried it, same error...